### ISSUE TITLE
MenuButton Allow Disabling AutoHide of Popupmenu icon area

### ISSUE THREAD

**Issue by Shadowblitz16** [godotengine/godot-proposals#8931]
### Describe the project you are working on

Super mario bros editor.

### Describe the problem or limitation you are having in your project

When trying to mimick windows 95 like gui I ran into an issue where the popup menus of menu button automatically hide
The icon/checkable area.

![image](https://github.com/godotengine/godot-proposals/assets/2453707/4ef32308-6f1f-46d3-8870-686a9f961117)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add Icon Hide mode enum to menu button and option button.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

It would basically be a enum which would be one of the following enum values...
- None
- One
- Both
- IconOnly
- CheckOnly

None will AutoHide if there isn't a icon or the menu item is not checkable
One will reserve one space for either icon or checkable
Both will reserve two spaces one for icon and one for checkable
IconOnly will reserve one space for icon only
CheckOnly will reserve one space for check only
 

Here is an example of one space reserved...
![image](https://github.com/godotengine/godot-proposals/assets/2453707/41b116d7-f015-425a-957d-adc6162e4f47)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Idk

### Is there a reason why this should be core and not an add-on in the asset library?

I icons and checkables are already part of core menu button and option button
This would just allow users to control how the menu item looks

**Comment by RedMser** [comment]
> Idk

As a workaround: Couldn't you add an empty icon (fully transparent texture) to one of the menu items to force it to reserve space?

**Comment by Shadowblitz16** [comment]
> > Idk
> 
> As a workaround: Couldn't you add an empty icon (fully transparent texture) to one of the menu items to force it to reserve space?

Yes that is a good idea. I might do that as a work around. thankyou!
